src/Form/PcscCsvActionForm.php

```diff
@@ -376,17 +420,30 @@ public function exportEnrollmentProducer() {
   /**
    * Helper function to build field enrollment data.
    *
+   * @param int|null $year
+   *   Optional year to filter by.
+   * @param int|null $quarter
+   *   Optional quarter to filter by.
+   *
    * @return array
    *   The data array.
    */
-  public function exportEnrollmentField() {
+  public function exportEnrollmentField(?int $year, ?int $quarter) {
+
+    // Build a quarter filter.
+    $quarter_filter = [];
+    if ($year && $quarter) {
+      $quarter_filter['pcsc_year'] = $year;
+      $quarter_filter['pcsc_quarter'] = $quarter;
+    }
+
     $data = [];
     foreach ($this->entities as $entity) {
       $farm_id = $entity->get('pcsc_farm_id')->value;
       $commodities = $this->entityTypeManager->getStorage('plan_record')->loadByProperties([
         'type' => 'pcsc_commodity',
         'plan' => $entity->id(),
-      ]);
+      ] + $quarter_filter);
       foreach ($commodities as $commodity) {
```

Comment:
The field enrollment export is the only one that _might_ have an issue with this filtering logic. The issue is that for this export we loop over all all `pcsc_commodity` records associated with the plan and filter by their `pcsc_year` and `pcsc_quarter`. For each commodity we grab the field it is associated with, but as implemented we are never sorting by the `pcsc_field` record's `pcsc_year` nor `pcsc_quarter`. It's not guaranteed that the field + commodity will have the same year + quarter.

That said, after thinking about this more, I think this is OK. The "Field Enrollment" export is really about the Commodity + Practice information associated with a Field. In fact, the Field's year + quarter are never used in the export. Maybe we don't need those fields - is this an oversight on our part?